examples/gno.land/r/demo/disperse/disperse.gno

```diff
@@ -0,0 +1,122 @@
+package disperse
+
+import (
+	"std"
+
+	tokens "gno.land/r/demo/grc20factory"
+)
+
+// Check if number of addresses and coins match
+func checkNumAddrValMatch(addresses []std.Address, coins std.Coins) {
+	if len(addresses) != len(coins) {
+		panic(ErrNumAddrValMismatch)
+	}
+}
+
+// Check if caller correctly sent positive amount of coins
+func checkCoinAmount(coins std.Coins) {
+	for _, coin := range coins {
+		if coin.Amount <= 0 {
+			panic(ErrNegativeCoinAmount)
+		}
+	}
+}
+
+// DisperseGnot parses receivers and amounts and sends out gnot
+func DisperseGnot(addresses []std.Address, coins std.Coins) {
+	coinSent := std.GetOrigSend() // get Coins sent with call
+	caller := std.GetOrigCaller() // get tx sender
+
+	checkCoinAmount(coins)
+	checkNumAddrValMatch(addresses, coins)
+
+	// Get address of Disperse realm
+	realmAddr := std.CurrentRealm().Addr()
+
+	// Get Banker
+	banker := std.GetBanker(std.BankerTypeOrigSend)
+
+	var totalAmount int64
+	for _, coin := range coins {
+		totalAmount += coin.Amount
+	}
+
+	// Check if total amount of coins sent is bigger or equal to total amount of coins to be sent
+	if totalAmount > coinSent.AmountOf("ugnot") {
+		panic(ErrWrongAmount)
+	}
+
+	// Check if the realm has zero ugnot
+	if banker.GetCoins(realmAddr).AmountOf("ugnot") < coinSent.AmountOf("ugnot") {
+		panic(ErrBalanceNotEqToAmountSent)
+	}
+
+	// Send coins
+	for i, _ := range addresses {
+		banker.SendCoins(realmAddr, addresses[i], std.Coins{coins[i]})
+	}
+
+	// Return possible leftover coins
+	for _, coin := range coinSent {
+		leftoverAmt := banker.GetCoins(realmAddr).AmountOf(coin.Denom)
+		if leftoverAmt > 0 {
+			send := std.Coins{{coin.Denom, leftoverAmt}}
+			banker.SendCoins(realmAddr, caller, send)
+		}
+	}
+}
+
+// DisperseToken disperses tokens to multiple addresses
+func DisperseToken(addresses []std.Address, coins std.Coins) {
```

Comment:
Interesting.

While it might be tempting to use std.Coins to represent GRC20 tokens, I don't think we should use it because it will create confusion on what is actually being sent (coins or grc20?). Let's switch to using the symbol + amount for GRC20. 
@moul, what do you think about this?


Also, please document that this function can only be called after approving disperse for the tokens you want to send.

Reply:
https://github.com/gnolang/gno/pull/2613/commits/dcd81fe1c62445f8924ea0b07716eb68adb079b9
https://github.com/gnolang/gno/pull/2613/commits/56d07169fe92cd10363165713edf9f0f88ba4958